ouroboros-network/src/Ouroboros/Network/Mux/Ingress.hs

```diff
@@ -67,28 +69,37 @@ negMiniProtocolMode ModeResponder = ModeInitiator
 
 {- | Decode a MuSDU header -}
 decodeMuxSDUHeader :: ProtocolEnum ptcl
-                   => BL.ByteString -> Maybe (MuxSDU ptcl)
+                   => BL.ByteString -> Either MuxError (MuxSDU ptcl)
 decodeMuxSDUHeader buf =
     case Bin.runGetOrFail dec buf of
-         Left  (_, _, _)  -> Nothing
-         Right (_, _, ph) -> Just ph
-
+         Left  (_, _, e)  -> Left $ MuxError MuxDecodeError e callStack
+         Right (_, _, ph) ->
+             let mode  = getMode $ mshIdAndMode ph
+                 mid_m = getId $ mshIdAndMode ph .&. 0x7fff in
+             case mid_m of
+                  Left  e   -> Left  $ MuxError MuxUnknownMiniProtocol ("id = " ++ show e) callStack
```

Comment:
I don't think the `callStack` is that useful when there are no `HasCallStack` constraints, but it might be useful when debugging.

Reply:
Thanks, I'll add the `HasCallStack` constraints.